src/V3EmitCSyms.cpp

```diff
@@ -185,6 +185,7 @@ class EmitCSyms final : EmitCBaseVisitorConst {
     }
 
     void varHierarchyScopes(string scp) {
+        string::size_type dotpos = scp.rfind('.');
```

Comment:
Please add comments on what's going on here.  It's potentially confusing that there's both string transformations in scopeSymString and here too.